charts/redpanda/chart_test.go

```diff
@@ -74,6 +84,39 @@ func TieredStorageSecretRefs(t *testing.T, secret *corev1.Secret) Values {
 	}
 }
 
+func TestTemplate(t *testing.T) {
+	ctx := testutil.Context(t)
+	client, err := helm.New(helm.Options{ConfigHome: testutil.TempDir(t)})
```

Comment:
If possible I would like to replace/inject the `charts/redpanda/Chart.lock` file with the following content

```
dependencies:
- name: console
  repository: https://charts.redpanda.com
  version: 0.7.21
- name: connectors
  repository: https://charts.redpanda.com
  version: 0.1.10
digest: sha256:020e143876af41c4cee1fe908bcd07151a4489115b0aa2100728a6ccc117671c
generated: "2024-03-11T15:50:33.503683+01:00"
```

I would recommend to perform `helm dep up` command before the for loop. That way tests would be deterministic.

Reply:
Fixed by running `helm dep build` in this test. We don't want this test to perform updates for us just to ensure that `charts` is in sync with `Chart.lock`.